src/blockchains/erc20/lib/chunk_utils.ts

```diff
@@ -0,0 +1,17 @@
+export function buildInclusiveChunks(start: number, end: number, chunkSize: number): [number, number][] {
+  if (chunkSize <= 0) {
+    throw new Error('chunkSize should be positive');
+  }
+
+  if (end < start) {
+    return [];
+  }
+
+  const chunks: [number, number][] = [];
+  for (let currentStart = start; currentStart <= end; currentStart += chunkSize) {
+    const currentEnd = Math.min(currentStart + chunkSize - 1, end);
```

Comment:
this check seems to be strange. worst case scenario we'll create a bunch of 0 chunks